### PR TITLE
Feature/curd

### PR DESCRIPTION
**Pull request by Artic13**
Linked to cloud Database - SupaBase
Implemented Create, Read, Update and Delete Functionality.
Created some helper functionality.
Tested and passed preliminary tests.

### CODE REVIEW

File: src/components/Testing.tsx [hunk 1]
```diff
@@ -16,27 +48,66 @@ export default function Testing(user : Struct.User) {
...
     console.log(Struct.timeCalc(dummyEvent[count]));
 
   }
-  // }
+  
+  async function testo(){
+    // const testl = new API();
+    // console.log( await testl.getEvent(1));
+    let dd : Struct.Event = {
+    id: 17,
```
Comment (hisnameisjoseph) [inline src/components/Testing.tsx:64]: where can I see this testing event?

File: src/components/Testing.tsx [hunk 2]
```diff
@@ -16,27 +48,66 @@ export default function Testing(user : Struct.User) {
   // function test (){
   //     console.log('this works yay!!');
   //     // number = (number + 10 ) %20;
+
+
   function LogEvent(){
```
Comment (hisnameisjoseph) [inline src/components/Testing.tsx:53]: What is this for? is this linked to supabase dummy event? it is showing import error from my side

**Review by hisnameisjoseph — COMMENTED** [review]
overall looks good to me

File: src/components/Testing.tsx (hunk 1 above)
Comment (Artic13) [reply src/components/Testing.tsx:64]: You will need to do checks in Supabase to see the data. Alternatively, you can test it with the other functions. For example, create (x), then call read(x) etc.

File: src/components/Testing.tsx (hunk 2 above)
Comment (Artic13) [reply src/components/Testing.tsx:53]: nvm, that's nothing important, just some dummy functions I did to conduct some testing, which will be removed before the project is complete. The testing.tsx file in mainly for my testing purposes.

**Comment by Artic13** [comment]
Main areas that should be reviewed are the main/page.tsx and utils/ServiceLayer.tsx files

**Review by hisnameisjoseph — APPROVED** [review]

**Pull request merged**